### PR TITLE
Unbreak hg-git

### PR DESCRIPTION
**Pull request by mikesperber**
###### Motivation for this change

hg-git did not work with the current Mercurial because of well-known issues,
one of which is fixed in the latest release, the second one of which has a simple
patch.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/hg-git/default.nix [hunk 1]
```diff
@@ -3,19 +3,39 @@
...
   src = fetchPypi {
     inherit pname version;
-    sha256 = "08kw1sj3sq1q1571hwkc51w20ks9ysmlg93pcnmd6gr66bz02dyn";
+    sha256 = "13hbm0ki6s88r6p65ibvrbxnskinzdz0m9gsshb8s571p91ymfjn";
   };
 
   propagatedBuildInputs = [ dulwich ];
+  patches = [
+    # Needs patch to work with Mercurial 4.8
+    # https://bitbucket.org/durin42/hg-git/issues/264/unexpected-keyword-argument-createopts-hg
```
Comment (Mic92) [inline pkgs/development/python-modules/hg-git/default.nix]: Can you use?

```nix
fetchpatch {
  url = "https://bitbucket.org/rsalmaso/hg-git/commits/a778506fd4be0bf1afa75755f6ee9260fa234a0f/raw";
  sha256 = "<insert-sha-sum>";
}
```

**Comment by mikesperber** [comment]
Sure, but could you briefly explain why? It seems it inserts a dependency on a bitbucket page making the build process more fragile.

**Comment by Mic92** [comment]
Then we don't need to carry the patch in our channel, it is clear where the patch comes from and our cache will also make sure that the patch is available when bitbucket is down.

**Comment by mikesperber** [comment]
Thanks for the explanation! (And thanks for the quick review.)

Done.

**Pull request merged**